libraries/USBDevice/USBSerial/CircBuffer.h

```diff
@@ -66,8 +56,8 @@ class CircBuffer {
 private:
     volatile uint16_t write;
     volatile uint16_t read;
-    uint16_t size;
-    T * buf;
+    static const int size = Size+1;  //a modern optimizer should be able to remove this so it uses no ram.
+    T buf[Size];
```

Comment:
The code used to malloc the equivalent of Size+1.  It seems like it should be "T buf[Size+1];" here.  Wouldn't the current code allow the read/write members to overflow the buf array by 1 element?